partner_property/models/res_company.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    partner_properties_definition = fields.PropertiesDefinition(
+        string="Partner Properties",
+        compute="_compute_partner_properties_definition",
+        inverse="_inverse_partner_properties_definition",
+    )
+
+    @api.depends_context("company")
+    def _compute_partner_properties_definition(self):
+        ICP = self.env["ir.config_parameter"]
```

Comment:
One thing, why don't you store this on the company itself? As it's right now, the properties are shared across all companies.

Reply:
Sorry, I thought we had already discussed that internally.

The option to make the partner_properties_definition field of res.company store was discarded because:

1- Set properties and values for a record in company 1 > OK.
![ejemplo-1](https://github.com/OCA/partner-contact/assets/4117568/4709b03a-4d85-4434-9fff-469ab327b63e)

2- Set properties and values for a record in company 2 > KO
![ejemplo-2](https://github.com/OCA/partner-contact/assets/4117568/5831502e-80ac-485a-8d11-c87c75e9c962)

In point 2 the values have **not** been saved, and when changing to company 1 the values have **disappeared**!
Another discarded option is to use the field **company_dependet=True** (odoo core has not taken into account this use case).

Having clarified all this, the idea is that the properties are defined at a "_global_" level (of all companies) and can be used in the context of any company.

Reply:
We have to keep current approach due to ORM limitations.